File: framework-global-id/src/main/java/com/github/caijh/framework/global/id/service/GlobalIdService.java
package com.github.caijh.framework.global.id.service;

import java.security.SecureRandom;
import java.util.List;

public interface GlobalIdService {

    default long generate(Long increase) {
        long saltA = new SecureRandom().nextLong(0L, 256L);
        long saltB = new SecureRandom().nextLong(0L, 256L);
        return increase << 16 | saltA << 8 | saltB;
    }

    long nextId();

    long nextId(String table);

    List<Long> nextIds(int n);

    List<Long> nextIds(String table, int n);

}

File: framework-global-id/src/main/kotlin/com/github/caijh/framework/globalid/service/GlobalIdService.kt
package com.github.caijh.framework.globalid.service

import kotlin.random.Random

interface GlobalIdService {

    fun generate(increase: Long): Long {
        val saltA = Random(System.currentTimeMillis()).nextLong(0L, 256L)
        val saltB = Random(System.currentTimeMillis()).nextLong(0L, 256L)
        return increase.shl(16) or (saltA.shl(8)) or saltB
    }

    fun nextId(): Long
}
